### PR TITLE
Support for Map<String, String> in configured builders.

### PR DESCRIPTION
**Pull request by tomas-langer**
### Description
Resovles #8217 

Adds explicit support for type `Map<String, String>`, and instead of doing the "complicated" get from config, it uses `config.detach().asMap()` to get the child values

### CODE REVIEW

**Review by Verdent — APPROVED** [review]

**Review by Tomas-Kraus — APPROVED** [review]
Looks good
